### PR TITLE
Scroller's bar can now be dragged to scroll content

### LINKED ISSUE #133: Cannot click or drag scroll bar
We don't yet have click and drag support to provide this functionality, postponing until we do

### PR DESCRIPTION
**Pull request by silvanocerza**
Fixes #133

### CODE REVIEW

File: widget/scroller.go [hunk 1]
```diff
@@ -26,34 +143,15 @@ func (s *scrollRenderer) updatePosition() {
...
-	return fyne.NewSize(theme.ScrollBarSize(), barHeight)
+	Renderer(s.vertBar).Layout(s.scroll.size)
 }
 
 func (s *scrollRenderer) Layout(size fyne.Size) {
 	c := s.scroll.Content
 	c.Resize(c.MinSize().Union(size))
+	// The scroll bar needs to be resized too
+	bar := s.vertBar
+	bar.Resize(bar.MinSize().Union(size))
```
Comment (andydotxyz) [inline widget/scroller.go]: This should not union with the main view Width - it should be only as wide as the theme.ScrollBarWidth() (probably the minSize.Width)
Also here we should move the bar into place as the scroll view should control it.

File: widget/scroller.go [hunk 2]
```diff
@@ -8,9 +8,126 @@ import (
...
+type scrollBarRenderer struct {
+	scrollBar *scrollBar
+	bar       *canvas.Rectangle
+
+	objects []fyne.CanvasObject
+}
+
+func (s *scrollBarRenderer) Layout(size fyne.Size) {
+	// Positions the bar on the far right of the container
+	s.scrollBar.Move(fyne.NewPos(size.Width-theme.ScrollBarSize(), 0))
```
Comment (andydotxyz) [inline widget/scroller.go]: I think this should be in the ScrollContainer renderer as it's our position within parent (noted the insertion point below)

File: widget/scroller.go [hunk 3]
```diff
@@ -71,7 +169,7 @@ func (s *scrollRenderer) Refresh() {
 }
 
 func (s *scrollRenderer) ApplyTheme() {
-	s.vertBar.FillColor = theme.ScrollBarColor()
+	Renderer(s.vertBar).ApplyTheme()
```
Comment (andydotxyz) [inline widget/scroller.go]: This should not be needed as children have this called trough the object tree

**Comment by silvanocerza** [comment]
Updated!

**Comment by andydotxyz** [comment]
Great thanks!

**Pull request merged**